magento/inventory
# Configurable stock status is not correct when using custom stock (MSI)

### Description (*)
This PR fix when MSI enabled to product ,the configurable product status must be updated when a child's product status is changed.

### Fixed Issues (if relevant)

1. Fixes #36154

https://github.com/magento/magento2/issues/36154

### Manual testing scenarios (*)

- create new source in admin > store > sources, i.e : source A
- create new stock in admin > store > stock, i.e : custom stock
- assign source A to Custom stock, and also assign custom stock to main website
- create new configurable product, i.e : test product
        - add simple product to the configurable product in configuration tab
        - assign simple product to source A
        - fill quantity, i.e : 1
- and save new created configurable product. (in this step the configurable product stock = In Stock)
- now change the simple product :
      - Source Item Status : from In Stock to Out Of Stock
      - Qty : from 1 to 0
      - save simple product
- check configurable product (the stock status is still In Stock, it must be Out Of Stock)



### Contribution checklist (*)
